Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -963,6 +963,8 @@ void EventsCodeGenerator::DeleteUselessEvents(gd::EventsList& events) {
  */
 void EventsCodeGenerator::PreprocessEventList(gd::EventsList& listEvent) {
   for (std::size_t i = 0; i < listEvent.GetEventsCount(); ++i) {
+    if (listEvent[i].IsDisabled()) continue;

```

Comment:
This is the fix. `Preprocess` would skip the event, but not the sub events with the lines below!